internal/proxy/proxy.go

```diff
@@ -582,6 +584,26 @@ func (p *Proxy) createRequestHash(url string, headers string) uint32 {
 	return h.Sum32()
 }
 
+func (p *Proxy) getUserWithRetry(repositoryCloneURL string, logEntry *log.Entry, retry int) (clients.Namespace, error) {
+	var namespace clients.Namespace
+	var error error
+
+	for i := 0; i < retry; i++ {
```

Comment:
No, it iterates retry times. Suppose you have retry=10 it iterates 0<10, 1<10, 2<10, 3<10, 4<10, 5<10, 6<10, 7<10, 8<10 and 9<10 Which is exactly 10 times

Reply:
and the last p.getUser will be 11th time - right?

Reply:
https://skillcrush.com/2013/01/17/why-programmers-start-counting-at-zero/

Reply:
Yes but it is outside the loop, and well It is just to reuse the last sleep as well. So inside loop is just 10 but technically it iterates +1 because of last one, but it is fine.

Also notice one stupid thing but true at the same time, we are saying that we want to retry 10 times, so the first one (the real one) should not be counted as retry 😆 so the code is fine.